timbermill-java/timbermill-local/src/main/java/com/datorama/oss/timbermill/ElasticsearchClient.java

```diff
@@ -942,11 +938,170 @@ long countByName(String name, String env) throws IOException {
     }
 
     public Bulker getBulker() {
-		return bulker;
-	}
+        return bulker;
+    }
+
+    public IndexRetryManager getRetryManager() {
+        return retryManager;
+    }
+
+    public void mergeIndexes() {
```

Comment:
potential very rare race condition, tell me if it make sense, not sure I would highly worry about it:
1. our existing (current) index is about to over
2. we generate new index to be used soon (start of rollout)
3. merge index is running: as the new index is not actually be used yet as our current alias (rollout in progress), then this index is found as potential index for merging?
4. rollout finish (this new index is being used now as current alias), while it also potentially participated in re-index

Reply:
spoke about this. The chance to have a race condition here is very slim